src/py-opentimelineio/opentimelineio/adapters/svg.py

```diff
@@ -0,0 +1,1017 @@
+# with the following modification; you may not use this file except in
+# compliance with the Apache License and the following modification to it:
+# Section 6. Trademarks. is deleted and replaced with:
+#
+# 6. Trademarks. This License does not grant permission to use the trade
+#    names, trademarks, service marks, or product names of the Licensor
+#    and its affiliates, except as required to comply with Section 4(c) of
+#    the License and to reproduce the content of the NOTICE file.
+#
+# You may obtain a copy of the Apache License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the Apache License with the above modification is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied. See the Apache License for the specific
+# language governing permissions and limitations under the Apache License.
+#
+
+"""OTIO to SVG Adapter
+   Points in calculations are y-up.
+   Points in SVG are y-down."""
+
+# otio
+import opentimelineio as otio
+
+# python
+import math
+from random import seed
+from random import random
+from collections import namedtuple
+
+COLORS = {
+    'transparent': (0, 0, 0, 0),
+    'black': (0.0, 0.0, 0.0, 255.0),
+    'white': (255.0, 255.0, 255.0, 255.0),
+    'transluscent_white': (255.0, 255.0, 255.0, 178.5),
+    'purple': (127.5, 0.0, 127.5, 255.0),
+    'light_blue': (134.895, 206.04, 235.11, 255.0),
+    'blue': (0.0, 0.0, 255.0, 255.0),
+    'dark_blue': (0.0, 0.0, 137.7, 255.0),
+    'green': (0.0, 127.5, 0.0, 255.0),
+    'dark_green': (0.0, 99.45, 0.0, 255.0),
+    'yellow': (255.0, 255.0, 0.0, 255.0),
+    'gold': (255.0, 214.2, 0.0, 255.0),
+    'orange': (255.0, 164.985, 0.0, 255.0),
+    'red': (255.0, 0.0, 0.0, 255.0),
+    'dark_red': (137.7, 0.0, 0.0, 255.0),
+    'brown': (137.7, 68.85, 25.5, 255.0),
+    'pink': (255.0, 191.25, 201.45, 255.0),
+    'gray': (127.5, 127.5, 127.5, 255.0),
+    'dark_gray': (168.3, 168.3, 168.3, 255.0),
+    'dark_gray_transluscent': (168.3, 168.3, 168.3, 200.0)
+}
+
+random_colors_used = []
+
+
+def get_random_color():
+    return [random(), random(), random()]
+
+
+def color_distance(c1, c2):
+    return sum([abs(x[0] - x[1]) for x in zip(c1, c2)])
+
+
+def generate_new_color(existing_colors):
+    max_distance = None
+    best_color = None
+    for i in range(0, 100):
+        color = get_random_color()
+        if not existing_colors:
+            return color
+        best_distance = min([color_distance(color, c) for c in existing_colors])
+        if not max_distance or best_distance > max_distance:
+            max_distance = best_distance
+            best_color = color
+    return best_color
+
+
+def random_color():
+    color = generate_new_color(random_colors_used)
+    random_colors_used.append(color)
+    return color[0] * 255.0, color[1] * 255.0, color[2] * 255.0, 255.0
```

Comment:
I had a Color class eariler, but since we just need to generate random colors and no other custom methods @ssteinbach suggested not using a class.
https://github.com/PixarAnimationStudios/OpenTimelineIO/pull/649#discussion_r389167101

Reply:
One of the things he mentioned in that previous comment was "not using any custom methods". Personally, I think between `color_distance`, `random_color`, and `svg_color`, there is probably enough to justify a class here.

@ssteinbach what do you think?